staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -434,6 +439,18 @@ func NewCacherFromConfig(config Config) (*Cacher, error) {
 		)
 	}()
 
+	// Check to see if cacher supports RequestWatchProgress and warn if it does not
+	parentCtx := context.Background()
+	// Create a context with cancellation
+	ctx, cancel := context.WithCancel(parentCtx)
+	defer cancel()
+
+	// No need to do when storage supports the feature, just log error
+	_, err := cacher.Supports(ctx, storage.RequestWatchProgress)
```

Comment:
Am I correct in understanding that the call to this method is not guaranteed to succeed? 
If so, then the call to `featuresupportchecker.SupportsCache` will always fail.

Reply:
As @serathius point out, i'm changing the Supports to always return cached value true if the feature is supported, We have two methods, 1. for checking if the feature is supported and caches the value(this will be something like `endpointSupports(ctx, client, endpoint, feature)`) 2. this `Supports` to only check if the cache is available and returns it. Currently i'm changing based on this comment https://github.com/kubernetes/kubernetes/pull/123264#discussion_r1502666061

Reply:
oh, ok, sorry I missed this comment.